nrf_802154_sl/include/fem/nrf_fem_protocol_api.h

```diff
@@ -0,0 +1,281 @@
+ * and the configuration is preserved between calls).
+ *
+ * The order of calls of this function and its `lna` counterpart must match the order of radio operations.
+ * I.e. if you want to perform the CCA and then send the frame, you need first to issue @ref nrf_802154_fal_lna_configuration_set and only after @ref nrf_802154_fal_pa_configuration_set.
+ *
+ * If a NRF_802154_PA_LNA_EVENT_TYPE_TIMER timer event is provided, the PA will be configured to activate or deactivate at the application-configured time gap
+ * before the timer instance reaches the given register_value. The time gap is set via @ref nrf_fem_interface_configuration_set.
+ *
+ * If a NRF_802154_PA_LNA_EVENT_TYPE_GENERIC event is provided, the PA will be configured to activate or deactivate when an event occurs.
+ *
+ * The function sets up the PPIs and the GPIOTE channel to activate PA for the upcoming radio transmission.
+ * The PA pin will be active until deactivated, which can happen either by encountering a configured deactivation event or by using @ref nrf_802154_fal_deactivate_now.
+ *
+ * @param[in] p_activate_event   Pointer to the activation event structure.
+ * @param[in] p_deactivate_event Pointer to the deactivation event structure.
+ *
+ * @pre To activate PA, nrf_fem_interface_configuration_set() must have been called first.
+ *
+ * @note If a timer event is provided, the caller of this function is responsible for starting the timer and configuring its shorts.
+ *       Moreover, the caller is responsible for stopping the timer no earlier than the compare channel of the lowest ID among the provided ones does expire.
+ *
+ * @retval   ::NRFX_SUCCESS               PA activate setup is successful.
+ * @retval   ::NRFX_ERROR_FORBIDDEN       PA is currently disabled.
+ * @retval   ::NRFX_ERROR_INVALID_STATE   PA activate setup could not be performed due to invalid or missing configuration parameters
+ *                                        in p_activate_event or p_deactivate_event, or both.
+ */
+int32_t nrf_802154_fal_pa_configuration_set(const nrf_802154_fal_event_t * const p_activate_event,
+                                            const nrf_802154_fal_event_t * const p_deactivate_event);
+
+/**
+ * @brief Clears up the configuration provided by the @ref nrf_802154_fal_pa_configuration_set function.
+ *
+ * @retval   ::NRFX_SUCCESS               PA activate setup purge is successful.
+ * @retval   ::NRFX_ERROR_FORBIDDEN       PA is currently disabled.
+ * @retval   ::NRFX_ERROR_INVALID_STATE   PA activate setup purge could not be performed due to invalid or missing configuration parameters,
+ *                                        that were previously used in @ref nrf_802154_fal_pa_configuration_set call.
+ */
+int32_t nrf_802154_fal_pa_configuration_clear(void);
+
+/**
+ * @brief Sets up LNA using the provided event for the upcoming radio reception.
```

Comment:
This is too much for doxygen doc in a header file, should be moved to RST documentation in the future. Make sure to follow up with @FrancescoSer